src/applications/vaos/tests/new-appointment/components/ReasonForAppointmentPage.unit.spec.jsx

```diff
@@ -1,189 +1,236 @@
+
+    fireEvent.click(
+      await screen.findByLabelText(/Routine or follow-up visit/i),
     );
+    const textBox = screen.getByRole('textbox');
+    fireEvent.change(textBox, { target: { value: 'test' } });
+    expect(textBox.value).to.equal('test');
 
-    form.unmount();
-  });
+    fireEvent.click(screen.getByText(/Continue/));
 
-  it('document title should match h1 text', () => {
-    const openReasonForAppointment = sinon.spy();
-    const updateReasonForAppointmentData = sinon.spy();
-    const pageTitle = 'Choose a reason for your appointment';
+    await waitFor(() =>
+      expect(screen.history.push.lastCall?.args[0]).to.equal(
+        '/new-appointment/choose-visit-type',
+      ),
+    );
 
-    const form = mount(
-      <ReasonForAppointmentPage
-        openReasonForAppointment={openReasonForAppointment}
-        updateReasonForAppointmentData={updateReasonForAppointmentData}
-        data={{}}
-      />,
+    fireEvent.click(
+      await screen.findByLabelText(/I have a new medical issue/i),
+    );
+    await waitFor(() =>
+      expect(screen.history.push.lastCall?.args[0]).to.equal(
+        '/new-appointment/choose-visit-type',
+      ),
     );
 
-    expect(form.find('h1').text()).to.equal(pageTitle);
-    expect(document.title).contain(pageTitle);
+    fireEvent.click(
+      await screen.findByLabelText(
+        /I have a concern or question about my medication/i,
+      ),
+    );
+    await waitFor(() =>
+      expect(screen.history.push.lastCall?.args[0]).to.equal(
+        '/new-appointment/choose-visit-type',
+      ),
+    );
 
-    form.unmount();
+    fireEvent.click(
+      await screen.findByLabelText(/My reason isn’t listed here/i),
+    );
+    await waitFor(() =>
+      expect(screen.history.push.lastCall?.args[0]).to.equal(
+        '/new-appointment/choose-visit-type',
+      ),
+    );
   });
 
-  it('should render radio buttons and a textarea', () => {
-    const openReasonForAppointment = sinon.spy();
-    const updateReasonForAppointmentData = sinon.spy();
-
-    const form = mount(
-      <ReasonForAppointmentPage
-        openReasonForAppointment={openReasonForAppointment}
-        updateReasonForAppointmentData={updateReasonForAppointmentData}
-        data={{}}
-      />,
+  it('should continue to the correct page for Community Care medical request', async () => {
+    const store = createTestStore(initialState);
+    await setTypeOfFacility(store, /Community Care/i);
+    const screen = renderWithStoreAndRouter(
+      <Route component={ReasonForAppointmentPage} />,
+      {
+        store,
+      },
     );
 
-    expect(form.find('input[type="radio"]').length).to.equal(4);
-    expect(form.find('textarea').length).to.equal(1);
+    expect(screen.baseElement).to.contain.text(
+      'Tell us the reason for this appointment',
+    );
 
-    form.unmount();
-  });
+    const textBox = screen.getByRole('textbox');
+    fireEvent.change(textBox, { target: { value: 'test' } });
```

Comment:
Since this text box is optional for CC, it would be good to have a test that continued to the next page without entering anything. I think you could repurpose this test for that.

Reply:
Makes sense